stdlib/src/collections/optional.mojo

```diff
@@ -193,6 +204,17 @@ struct Optional[T: CollectionElement](
             return False
         return not rhs
 
+    fn __ne__(self, rhs: NoneType) -> Bool:
```

Comment:
**Suggestion** I think `opt == None` will work if you change this signature to:

```mojo
    # TODO(MSTDL-715):
    #   This initializer should not be necessary, we should need
    #   only the initilaizer from a `NoneType`.
    fn __ne__(self, rhs: NoneType._mlir_type):
```

(See the similar comment and pattern on `Optional.__init__(inout self, value: NoneType._mlir_type)`.)

Reply:
I just tried and got the following error:

```
/projects/open_source/mojo/stdlib/test/collections/test_optional.mojo:148:5: error: cannot bind MLIR type 'None' to trait 'CollectionElement'
def test_optional_equality():
    ^
/projects/open_source/mojo/stdlib/src/builtin/value.mojo:148:8: note: MLIR type cannot satisfy required trait function here
    fn __init__(inout self, *, other: Self):
       ^
/projects/open_source/mojo/stdlib/test/collections/test_optional.mojo:148:5: error: cannot bind MLIR type 'None' to trait 'CollectionElement'
def test_optional_equality():
    ^
/projects/open_source/mojo/stdlib/src/builtin/value.mojo:148:8: note: MLIR type cannot satisfy required trait function here
    fn __init__(inout self, *, other: Self):
       ^
```
Since I get the same error when I didn't do any modifications to `Optional` I believe that it's a bug in the overload resolution.